src/services/addAddresses.ts

```diff
@@ -0,0 +1,52 @@
+import { Customer } from '../types/user/customer';
+import { FormData } from '../types/user/formData';
+
+export const addAddresses = async (data: FormData, customer: Customer, userToken: string) => {
+  console.log(customer);
+
+  const isShippingDefaultAddress = data.isShippingDefaultAddress;
+  const isBillingDefaultAddress = data.isBillingDefaultAddress;
+  const isBillingSameAsShipping = data.isBillingSameAsShipping;
+
+  const shippingId = customer.addresses[0].id;
+  const billingId = data.isBillingSameAsShipping ? 1 : customer.addresses[1].id;
+
+  const actions = [
+    { action: 'addShippingAddressId', addressId: shippingId },
+    { action: 'addBillingAddressId', addressId: billingId },
+  ];
+
+  if (isBillingSameAsShipping) {
+    actions.forEach((item) =>
+      item.action === 'addBillingAddressId' ? (item.addressId = shippingId) : null,
+    );
+  }
+
+  if (isShippingDefaultAddress) {
+    actions.push({ action: 'setDefaultShippingAddress', addressId: shippingId });
+  }
+  if (isBillingDefaultAddress) {
+    if (data.isBillingSameAsShipping) {
+      actions.push({ action: 'setDefaultBillingAddress', addressId: shippingId });
+    } else {
+      actions.push({ action: 'setDefaultBillingAddress', addressId: billingId });
+    }
```

Comment:
Буквально одно и то же, только одно значение меняется 🙂 
Может лучше сделать через тернарный оператор?
Например вот так:
```suggestion
actions.push({ action: 'setDefaultBillingAddress', addressId: data.isBillingSameAsShipping ? shippingId : billingId });
```

Reply:
Согласен. Сделал через тернарный оператор.